samples/client/petstore/c/api/petAPI.c

```diff
@@ -0,0 +1,51 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include "apiClient.h"
+#include "cJSON.h"
+#include "pet.h"
+
+#define MAX_BUFFER_LENGTH 9
+
+pet_t *petApi_getPetById(long petId) {
+	pet_t *pet;
+	apiClient_t *apiClient;
+	char *petIdString = malloc(MAX_BUFFER_LENGTH);
+
+	snprintf(petIdString, MAX_BUFFER_LENGTH, "%li", petId);
+
+	apiClient = apiClient_create();
+	apiClient_invoke(apiClient,
+	                 "pet",
+	                 petIdString,
+	                 NULL);
+	pet = pet_parseFromJSON(apiClient->dataReceived);
+	if(pet == NULL) {
+		return 0;
+	} else {
+		cJSON *jsonObject = pet_convertToJSON(pet);
+		cJSON_Delete(jsonObject);
+	}
+	apiClient_free(apiClient);
```

Comment:
@PowerOfCreation would it be possible to reuse the same `apiClient` across different operations/methods (e.g. `petApi_addPet`)?

We may also want to store authentication-related information (e.g HTTP username, password, API key, etc) in `apiClient`

Reply:
I have tried making it thread safe, but I could make a global variable apiClient too and reuse that in all methods. I will take a look, whether I can find a way to make it thread safe and reuse the apiClient.

Authentication related data can be stored in the apiClient when I have changed it, so it won't get created/deleted each time an operation is being performed.